### PR TITLE
Ripple triggers twice on mobile

### PR DESCRIPTION
**Pull request by aluminick**

### CODE REVIEW

**Comment by aluminick** [comment]
The anchor tag button is not working because of the preventDefault inside `onTouchEnd`. I added it because for some reason it triggers the `onMouseDown`. So the workaround I used is what you told me to check if one of them is activated then don't trigger the other one.

Removing the RAF on all test cases works fine except, the checking of component if it has the `mdc-ripple-upgraded` class. Maybe because it's async. Wrapping the assertion with `setTimeout` or `setImmediate` fixes this. However i'm not sure is this is the right solution.

File: packages/ripple/index.tsx [hunk 1]
```diff
@@ -242,6 +242,7 @@ export function withRipple <
   };
 
   handleTouchEnd = (e: React.TouchEvent<Surface>) => {
+    e.preventDefault();
```
[the remark this answers is not in the document]
Comment (moog16) [reply packages/ripple/index.tsx]: could you please add that comment next to the `e.preventDefault();` line

**Comment by moog16** [comment]
@aluminick we shouldn't use setTimeout when there isn't one in the code. I think we actually need a rAF since there is an rAF in MDC Web's `activate_` method. Sorry for making you do extra work - lets just leave the rAFs for now.

**Comment by moog16** [comment]
@aluminick changes look awesome! Please fix `npm run lint` and we should be good to go pending the screenshots pass.

**Comment by aluminick** [comment]
@moog16 no worries. I enjoy debugging.

**Comment by moog16** [comment]
tests: https://github.com/material-components/material-components-web-react/pull/659

**Comment by moog16** [comment]
@aluminick please sign PR

**Comment by aluminick** [comment]
I signed it

**Pull request merged**